sql/util_time.sql

```diff
@@ -81,7 +81,7 @@ $BODY$
     _timescaledb_catalog.continuous_agg cagg
     LEFT JOIN _timescaledb_catalog.continuous_aggs_completed_threshold completed ON completed.materialization_id = cagg.mat_hypertable_id
   WHERE
-    cagg.raw_hypertable_id = $1;
+    cagg.mat_hypertable_id = $1;
```

Comment:
I think it is a re-occurring problem that we expose internal identifiers in function interfaces, even if we're dealing with internal functions. I think the logical thing here is for this function to take the `regclass` (note, not `oid`) of the continuous aggregate view, which is the object that represents the continuous aggregate.  The current `hypertable_id` parameter is very ambiguous as it is unclear what ID this refers to given the multiple hypertables (raw, mat) that are associated with the cagg. It is very easy to make a mistake and the function doesn't give any help beyond just standard SQL error in case of a mismatch. Describing the view query would also make a lot more sense, compare this

```
_timescaledb_internal.to_timestamp_without_timezone(_timescaledb_internal.cagg_watermark(25)
```

to this:
```
_timescaledb_internal.to_timestamp_without_timezone(_timescaledb_internal.cagg_watermark('conditions_2h')
```

At the very least, we should make it clear from the argument parameter which table `regclass` this should be, and raise appropriate errors in case the `regclass` passed in is not the table type we expect. Currently, `hypertable_id` really means nothing in this context.

Reply:
Point noted (cc @svenklemm).  As discussed at the meeting, no API changes for this PR.